lib/spack/docs/build_systems/makefilepackage.rst

```diff
@@ -0,0 +1,295 @@
+       ]
+
+
+``cloverleaf`` is a good example of a package that uses this strategy.
+
+"""""""""""""
+Edit Makefile
+"""""""""""""
+
+Some Makefiles are just plain stubborn and will ignore command-line
+variables. The only way to ensure that these packages build correctly
+is to directly edit the Makefile. Spack provides a ``FileFilter`` class
+and a ``filter_file`` method to help with this. For example:
+
+.. code-block:: python
+
+   def edit(self, spec, prefix):
+       makefile = FileFilter('Makefile')
+
+       makefile.filter('CC = gcc',  'CC = cc')
+       makefile.filter('CXX = g++', 'CC = c++')
+
+
+``stream`` is a good example of a package that involves editing a
+Makefile to set the appropriate variables.
+
+"""""""""""
+Config file
+"""""""""""
+
+More complex packages often involve Makefiles that _include_ a
+configuration file. These configuration files are primarily composed
+of variables relating to the compiler, platform, and the location of
+dependencies or names of libraries. Since these config files are
+dependent on the compiler and platform, you will often see entire
+directories of examples for common compilers and architectures. Use
+these examples to help determine what possible values to use.
+
+If the config file is long and only contains one or two variables
+that need to be modified, you can use the technique above to edit
+the config file. However, if you end up needing to modify most of
+the variables, it may be easier to write a new file from scratch.
+
+If each variable is independent of each other, a dictionary works
+well for storing variables:
+
+.. code-block:: python
+
+   def edit(self, spec, prefix):
+       config = {
+           'CC': 'cc',
+           'MAKE': 'make',
+       }
+
+       if '+blas' in spec:
+           config['BLAS_LIBS'] = spec['blas'].libs.joined()
+
+       with open('make.inc', 'w') as inc:
+           for key in config:
+               inc.write('{0} = {1}\n'.format(key, config[key]))
+
+
+``elk`` is a good example of a package that uses a dictionary to
+store configuration variables.
+
+If the order of variables is important, it may be easier to store
+them in a list:
+
+.. code-block:: python
+
+   def edit(self, spec, prefix):
+       config = [
+           'INSTALL_DIR = {0}'.format(prefix),
+           'INCLUDE_DIR = $(INSTALL_DIR)/include',
+           'LIBRARY_DIR = $(INSTALL_DIR)/lib',
+       ]
+
+       with open('make.inc', 'w') as inc:
+           for var in config:
+               inc.write('{0}\n'.format(var))
+
+
+``hpl`` is a good example of a package that uses a list to store
+configuration variables.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Variables to watch out for
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following is a list of common variables to watch out for:
+
+* **Compilers**
+
+  This includes variables such as ``CC``, ``CXX``, ``F77``, ``F90``,
+  and ``FC``, as well as variables related to MPI compiler wrappers,
+  like ``MPICC`` and friends.
```

Comment:
Instead of "and friends", just list them out.  I know it's tedious, but it makes the docs more useful.